uaclient/contract.py

```diff
@@ -468,7 +556,22 @@ def _get_activity_info(self):
             "desktop": system.is_desktop(),
             "virt": system.get_virt_type(),
             "clientVersion": version.get_version(),
-            "cpu_type": cpu_type.get_cpu_type(),
+            "cpu_type": CPUTypeData(
+                cpuinfo_cpu=cpuinfo.cpuinfo_cpu,
+                cpuinfo_cpu_architecture=cpuinfo.cpuinfo_cpu_architecture,
+                cpuinfo_cpu_family=cpuinfo.cpuinfo_cpu_family,
+                cpuinfo_cpu_implementer=cpuinfo.cpuinfo_cpu_implementer,
+                cpuinfo_cpu_part=cpuinfo.cpuinfo_cpu_part,
+                cpuinfo_cpu_revision=cpuinfo.cpuinfo_cpu_revision,
+                cpuinfo_cpu_variant=cpuinfo.cpuinfo_cpu_variant,
+                cpuinfo_model=cpuinfo.cpuinfo_model,
+                cpuinfo_model_name=cpuinfo.cpuinfo_model_name,
+                cpuinfo_stepping=cpuinfo.cpuinfo_stepping,
+                cpuinfo_vendor_id=cpuinfo.cpuinfo_vendor_id,
+                sys_firmware_devicetree_base_model=cpuinfo.sys_firmware_devicetree_base_model,  # noqa: E501
+                sysinfo_model=cpuinfo.sysinfo_model,
+                sysinfo_type=cpuinfo.sysinfo_type,
+            ).to_dict(keep_none=False),
```

Comment:
@orndorffgrant if we do `keep_none=False` we will discard some keys depending on the architecture right ? Shouldn't this value be true ?

Reply:
Double checked with @orndorffgrant and the Contract Server treats a non-existent field the same as a field with a null value